### PR TITLE
278 technical learning path view minimap better visibility

### PR DESCRIPTION
**Pull request by KLee954**
## Description
Changed the style of the React Flow Minimap and added resizing behaviour. 

## Type of changes

- [ ] Bugfix 
- [x] Feature 
- [ ] Refactoring 
- [ ] Optimization 
- [ ] Documentation 
- [ ] CI/CD 
- [ ] Other 

## Requirements
- [ ] Everything build out of Material UI components
- [ ] New Material UI components capsulated via dependency injection in "@common/components"
- [ ] Provide basic information pieces in comments
- [ ] Establish interface to backend (if necessary) 
- [ ] Use Skeleton components if something has to be fetched (if necessary)
- [ ] Error management: Components should handle undefined inputs
- [x] Components should be resuable (if possible)
- [ ] [Component Requirements](https://lab.las3.de/nextcloud/index.php/apps/onlyoffice/510074?filePath=%2FHASKI-Extern%2F06-Frontend%2F03-UX%2F04-Implementation%2FComponent_Requirements.docx) are met

### CODE REVIEW

**Comment by KLee954** [comment]
newUI

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.test.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import '@testing-library/jest-dom'
+import { act, fireEvent, render } from '@testing-library/react'
+import { mockReactFlow } from '@mocks'
+import { ReactFlow } from 'reactflow'
+import WrappedMiniMap from './ResponsiveMiniMap'
+
+describe('WrappedMiniMap component', () => {
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.test.tsx:7]:
```suggestion
describe('ResponsiveMiniMap tests', () => {
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.test.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  beforeEach(() => {
+    mockReactFlow()
+  })
+  it('renders', () => {
+    const { getByTestId } = render(
+      <ReactFlow>
+        <WrappedMiniMap />
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.test.tsx:14]:
```suggestion
        <ResponsiveMiniMap />
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.test.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      </ReactFlow>
+    )
+    expect(getByTestId('rf__minimap')).toBeInTheDocument()
+  })
+
+  it('resizes according to window size', () => {
+    const { getByTestId } = render(
+      <ReactFlow>
+        <WrappedMiniMap />
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.test.tsx:23]:
```suggestion
        <ResponsiveMiniMap />
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    const scalerange = 0.6
+    const translationrange = 2.5
+    const translationoffest = 1
+    const scaling = scaleoffset + scalerange * Math.tanh(window.innerWidth / theme.breakpoints.values.xl - scaleoffset)
+    const translation =
+      -translationrange * Math.tanh(window.innerWidth / theme.breakpoints.values.xl - translationoffest)
+    return { transform: `scale(${scaling}) translate(${translation}rem, ${translation}rem)` }
+  }, [])
+
+  const [minimapSize, setMiniMapSize] = useState(getMapTransform())
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx:29]: CodeSmell
```suggestion
  const [miniMapSize, setMiniMapSize] = useState(getMapTransform())
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  useEffect(() => {
+    window.addEventListener('resize', handleResize)
+    return () => {
+      window.removeEventListener('resize', handleResize)
+    }
+  }, [handleResize])
+
+  return (
+    <MiniMap
+      style={minimapSize}
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx:44]:
```suggestion
      style={miniMapSize}
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      nodeColor={theme.palette.primary.light}
+      maskStrokeColor="black"
+      maskStrokeWidth={2}
+      nodeStrokeColor={theme.palette.primary.main}
+      nodeStrokeWidth={5}
+    />
+  )
+}
+
+export default memo(WrappedMiniMap)
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx:55]:
```suggestion
export default memo(ResponsiveMiniMap)
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * WrappedMiniMap component.
+ *
+ * @remarks
+ * WrappedMiniMap adds a resize functionality to the reactflow minimap.
+ * It can only be used inside a reactflow component to display a minimap of the flowchart.
+ *
+ * @category Components
+ */
+const WrappedMiniMap = () => {
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx:14]:
```suggestion
const ResponsiveMiniMap = () => {
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import { MiniMap } from 'reactflow'
+import { useTheme } from '@common/hooks'
+import { useEffect, useState, useCallback, memo } from 'react'
+
+/**
+ * WrappedMiniMap component.
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx:6]:
```suggestion
 * ResponsiveMiniMap component.
```

File: src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * @remarks
+ * WrappedMiniMap adds a resize functionality to the reactflow minimap.
```
Comment (juli-txt) [inline src/components/ResponsiveMiniMap/ResponsiveMiniMap.tsx:9]:
```suggestion
 * ResponsiveMiniMap adds a resize functionality to the reactflow minimap.
```

**Review by juli-txt — APPROVED** [review]

**Pull request merged**